luarules/gadgets/unit_evolution.lua

```diff
@@ -95,7 +95,15 @@ if gadgetHandler:IsSyncedCode() then
 
 		-- },
 
-
+    function ReturnToBase(unitID)
+		local x,y,z = spGetUnitPosition(unitID) -- usefull if you want to spawn explosions or other effects where you were.
+		local team = spGetUnitTeam(unitID)
+		local ox,oy,oz = Spring.GetTeamStartPosition(team)
+		Spring.SetUnitPosition(unitID, ox, oz, true)
```

Comment:
what if that spot got nuked and is no longer pathable? what if there's a huge afus farm there and you clip into a building

Reply:
This modification isn't anywhere near finished.

But, you raise some really good points I hadn't considered how to tackle yet. Are there any widgets or gadgets you can think of that solve a similar problem I can draw inspiration from?

Reply:
There is some transport unload code that does some collision avoidance, though I'm not sure how well it will translate to your use case.